File: examples/python/aeotecdsb09104.py
#!/usr/bin/python
# Author: Jon Trulson <jtrulson@ics.com>
# Copyright (c) 2016 Intel Corporation.
#
# Permission is hereby granted, free of charge, to any person obtaining
# a copy of this software and associated documentation files (the
# "Software"), to deal in the Software without restriction, including
# without limitation the rights to use, copy, modify, merge, publish,
# distribute, sublicense, and/or sell copies of the Software, and to
# permit persons to whom the Software is furnished to do so, subject to
# the following conditions:
#
# The above copyright notice and this permission notice shall be
# included in all copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.

from __future__ import print_function
import time, sys, signal, atexit
from upm import pyupm_ozw as sensorObj

def main():
    # This function lets you run code on exit
    def exitHandler():
        print("Exiting")
        sys.exit(0)

    # Register exit handlers
    atexit.register(exitHandler)

    defaultDev = "/dev/ttyACM0"
    if (len(sys.argv) > 1):
        defaultDev = sys.argv[1]
    print("Using device", defaultDev)

    # Instantiate an Aeotec DSB09104 instance, on device node 12.  You
    # will almost certainly need to change this to reflect your own
    # network.  Use the ozwdump example to see what nodes are available.
    sensor = sensorObj.AeotecDSB09104(12)

    # The first thing to do is create options, then lock them when done.
    sensor.optionsCreate()
    sensor.optionsLock()

    # Next, initialize it.
    print("Initializing, this may take awhile depending on your ZWave network")

    sensor.init(defaultDev)
    print("Initialization complete")

    print("Querying data...")

    while (True):
        sensor.update()

        print("Watts, Channel 1: %0.03f W" % sensor.getWattsC1())
        print("Watts, Channel 2: %0.03f W" % sensor.getWattsC2())
        print("Watts, Channel 3: %0.03f W" % sensor.getWattsC3())

        print("Energy, Channel 1: %0.03f kWh" % sensor.getEnergyC1())
        print("Energy, Channel 2: %0.03f kWh" % sensor.getEnergyC2())
        print("Energy, Channel 3: %0.03f kWh" % sensor.getEnergyC3())

        print("Battery Level: %d\n" % sensor.getBatteryLevel())
        time.sleep(3)

if __name__ == '__main__':
    main()

File: examples/javascript/aeotecdsb09104.js
/*
 * Author: Jon Trulson <jtrulson@ics.com>
 * Copyright (c) 2016 Intel Corporation.
 *
 * This program and the accompanying materials are made available under the
 * terms of the The MIT License which is available at
 * https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 */


var sensorObj = require('jsupm_ozw');

/************** Main code **************/
var defaultDev = "/dev/ttyACM0";

// if an argument was specified, use it as the device instead
if (process.argv.length > 2)
{
    defaultDev = process.argv[2];
}
console.log("Using device " + defaultDev);

// Instantiate an Aeotec DSB09104 instance, on device node 12.  You
// will almost certainly need to change this to reflect your own
// network.  Use the ozwdump example to see what nodes are
// available.
var sensor = new sensorObj.AeotecDSB09104(12);

// The first thing to do is create options, then lock them when done.
sensor.optionsCreate();
sensor.optionsLock();

// Next, initialize it.
console.log("Initializing, this may take awhile depending on your ZWave network");

sensor.init(defaultDev);
console.log("Initialization complete");

console.log("Querying data...");

setInterval(function()
{
    sensor.update();

    console.log("Watts, Channel 1: "
                + sensor.getWattsC1()
                + " W");

    console.log("Watts, Channel 2: "
                + sensor.getWattsC2()
                + " W");

    console.log("Watts, Channel 3: "
                + sensor.getWattsC3()
                + " W");

    console.log("Energy, Channel 1: "
                + sensor.getEnergyC1()
                + " kWh");

    console.log("Energy, Channel 2: "
                + sensor.getEnergyC2()
                + " kWh");

    console.log("Energy, Channel 3: "
                + sensor.getEnergyC3()
                + " kWh");

    console.log("Battery Level: "
                + sensor.getBatteryLevel()
                + "%");

    console.log();
}, 3000);


// exit on ^C
process.on('SIGINT', function()
{
    sensor = null;
    sensorObj.cleanUp();
    sensorObj = null;
    console.log("Exiting.");
    process.exit(0);
});
